### ISSUE TITLE
Adding custom options / upsell products drop some product properties

### ISSUE THREAD

**Issue by vincentwinkel** [magento/magento2#8889]
### Preconditions
1. Magento CE 2.0.13 without sample data is installed
2. Also tested on CE 2.0.9
3. PHP 5.6.23 CLI

### Steps to reproduce
1. Create a custom Console Command
2. Add this code :
```php
foreach($_data['product'] as $productId => $doesnt_matter) {
	$this->_output->write('Add options for product'.$productId.'...');
	$product=$this->_objectManager->create(
		'\Magento\Catalog\Model\Product'
	)->load($productId);
	$options=[
		[
			'title' => 'prescription_id'
			'type' => 'field'
			'is_require' => false
			'sort_order' => 1
			'price' => 0
			'price_type' => fixed
		],
		[
			'title' => 'diagnostic'
			'type' => 'radio'
			'is_require' => false
			'sort_order' => 2
			'values' => [
				[
					[title] => 'no_correction'
					[price] => 10
					[price_type] => 'fixed'
					[sort_order] => 1
				],
				[
					[title] => 'myopia'
					[price] => 20
					[price_type] => 'fixed'
					[sort_order] => 2
				],
				[
					[title] => 'hyperopia'
					[price] => 30
					[price_type] => 'fixed'
					[sort_order] => 3
				],
				[
					[title] => 'progressive'
					[price] => 40
					[price_type] => 'fixed'
					[sort_order] => 4
				]
			]
		]
		//etc...
	];
	$product//->setStoreId(0)
		->setCanSaveCustomOptions(true)
		->setProductOptions($options) //Test 1
		//->setOptions($options) //Test 2
		//->setCustomOptions($options) //Test 3
		->save();
	$this->_output->writeln(' ok');
}
```
3. Execute the command

### Expected result
1. See Custom options in Backend (Catalog > Product > Custom Options)

### Actual result
1. If I uncomment setStoreId(0), all product names, prices, visibility, sku, quantities become empty and custom options aren't added
2. If I comment setStoreId(), custom options are added but prices become empty

This problem about conflict between product properties is also visible with this piece of code (maybe I must create a new issue ?) :
```php
foreach($_data['related'] as $productId => $relatedData) {
	//$relatedData = [
	//	4169 =>[
	//		'position' => 1
	//	]
	//]
	//for instance
	$this->_output->write('Add related product link for '.$productId.'...');
	$product=$this->_objectManager->create(
		'\Magento\Catalog\Model\Product'
	)->load($productId);
	$product->setUpSellLinkData($relatedData)->save(); //BUG HERE: all prices dropped from database
	$this->_output->writeln(' ok');
}
```

**Comment by vincentwinkel** [comment]
I found an alternative for links.
With the same base data example :
```php
//Non-working example
//for instance: $relatedData = [
//	4169 =>[
//		'position' => 1
//	]
//]
$product=$this->_objectManager->create(
	'\Magento\Catalog\Model\Product'
)->load($productId);
$product->setRelatedLinkData($relatedData)->save(); //Doesn't work: price removed from DB

//Working example
//for instance: $relatedData = [
//	'product_id' => 4169
//	'linked_product_id' => 4170
//	'link_type_id' => 1
//]
$this->_connection->query(
	'INSERT INTO catalog_product_link'
	.' (product_id,linked_product_id,link_type_id)'
	.' VALUES ('
	.''.$relatedData['product_id'].','
	.''.$relatedData['linked_product_id'].','
	.''.$relatedData['link_type_id']
	.')'
);
```
This alternative also works for `$product->setUpSellLinkData()` (tested), and I guess for `$product->setCrossSellLinkData()` (not tested). But how can I do to add custom options to products ? I don't want to use raw SQL queries, it's too complex !

**Comment by magento-engcom-team** [comment]
@vincentwinkel, thank you for your report.
The issue is already fixed in 2.2.0
